homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -70,48 +70,29 @@ def get_new_prayer_times(self):
     async def async_schedule_future_update(self):
```

Comment:
If the update will be on a specific time this method is no longer required.

Reply:
It's on a specific time (02:01 AM) but it can be on either the next day or the same day (handles if the integration was started/the server restarted/etc before 0200AM in the morning).

Do you think it is still good to handle this case?

Reply:
The timings are fetched on initial setup or restart, you only need to schedule it the next day.

Reply:
This is a corner case: suppose the restart/initialization happen at 01:00AM on Nov 6th, if schedule the next update for 02:01AM the next day (Nov 7th), then the timing will be incorrect for the entire day of Nov 6th (which happens to be the day when DST changes in some places).

Reply:
DST change is happening in less than 2 weeks and this issue has been here for a long time. Do you have a fix?

Reply:
> The timings are fetched on initial setup or restart, you only need to schedule it the next day.

This is a home automation system, a user shouldn't be concerned with having to manually restart the system to fix anything